hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -4566,7 +4586,7 @@ private void regionOffline(final HRegionInfo regionInfo, final State state) {
     // Tell our listeners that a region was closed
     sendRegionClosedNotification(regionInfo);
     // also note that all the replicas of the primary should be closed
-    if (state != null && state.equals(State.SPLIT)) {
+    if (force || state != null && state.equals(State.SPLIT)) {
```

Comment:
Recommended to add a set of parenthesis here, makes intent clearer:

`(force || (state != null && state.equals(State.SPLIT)))`

Reply:
good catch, updated.